winml/lib/Api.Core/WinMLAdapter.cpp

```diff
@@ -0,0 +1,840 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "inc/WinMLAdapter.h"
+#include "inc/CustomRegistryHelper.h"
+#include "PheonixSingleton.h"
+#include "inc/LotusEnvironment.h"
+#include "inc/AbiCustomRegistryImpl.h"
+#include "core/providers/dml/DmlExecutionProvider/inc/DmlExecutionProvider.h"
+#include "core/providers/dml/GraphTransformers/GraphTransformerHelpers.h"
+#include "core/providers/dml/OperatorAuthorHelper/SchemaInferenceOverrider.h"
+
+#include "LearningModelDevice.h"
+#include "TensorFeatureDescriptor.h"
+#include "ImageFeatureDescriptor.h"
+#include "api.image/inc/D3DDeviceCache.h"
+
+#include "PheonixSingleton.h"
+
+#include "DmlOrtSessionBuilder.h"
+#include "CpuOrtSessionBuilder.h"
+
+#include <io.h>
+#include <fcntl.h>
+
+#include "ZeroCopyInputStreamWrapper.h"
+#include "google/protobuf/io/zero_copy_stream_impl.h"
+
+#include "FeatureDescriptorFactory.h"
+
+
+using namespace winrt::Windows::AI::MachineLearning;
+
+namespace Windows::AI::MachineLearning::Adapter {
+
+// ORT intentionally requires callers derive from their session class to access
+// the protected Load method used below.
+class InferenceSessionProtectedLoadAccessor : public onnxruntime::InferenceSession {
+ public:
+  onnxruntime::common::Status
+  Load(std::unique_ptr<ONNX_NAMESPACE::ModelProto> p_model_proto) {
+    return onnxruntime::InferenceSession::Load(std::move(p_model_proto));
+  }
+};
+
+
+
+// class AbiSafeTensor
+//
+class AbiSafeTensor : public Microsoft::WRL::RuntimeClass <
+    Microsoft::WRL::RuntimeClassFlags<Microsoft::WRL::ClassicCom>,
+    ITensor> {
+private:
+    onnxruntime::Tensor& tensor_;   // weak ref
+    ComPtr<IOrtValue> value_;       // strong ref
```

Comment:
I believe our code convention puts data members at the end of a class, right?

Reply:
probably.    i changed it to the top when the classes are coded inside the CPP file (also not something we normally do) .  when you put them at the bottom they are much harder to find.

---
In reply to: [347044267](https://github.com/microsoft/onnxruntime/pull/2406#discussion_r347044267) [](ancestors = 347044267)